__tests__/main.test.ts

```diff
@@ -0,0 +1,116 @@
+import { run } from "../src/labeler";
+import { GitHub } from "@actions/github";
+import * as core from "@actions/core";
+
+const fs = jest.requireActual("fs");
+
+jest.mock("@actions/core");
+jest.mock("@actions/github");
+
+const gh = new GitHub("_");
+const addLabelsMock = jest.spyOn(gh.issues, "addLabels");
+const removeLabelMock = jest.spyOn(gh.issues, "removeLabel");
+const reposMock = jest.spyOn(gh.repos, "getContents");
+const paginateMock = jest.spyOn(gh, "paginate");
+const getPullMock = jest.spyOn(gh.pulls, "get");
+
+const yamlFixtures = {
+  "only_pdfs.yml": fs.readFileSync("__tests__/fixtures/only_pdfs.yml"),
+};
+
+function usingLabelerConfigYaml(fixtureName: keyof typeof yamlFixtures): void {
+  reposMock.mockResolvedValue(<any>{
+    data: { content: yamlFixtures[fixtureName], encoding: "utf8" },
+  });
+}
+
+function mockGitHubResponseChangedFiles(...files: string[]): void {
+  const returnValue = files.map((f) => ({ filename: f }));
+  paginateMock.mockReturnValue(<any>returnValue);
+}
+
+afterAll(() => jest.restoreAllMocks());
```

Comment:
Nit: consider setting the jest config value: https://jestjs.io/docs/configuration#restoremocks-boolean

Like we do for `clearMocks` https://github.com/actions/labeler/blob/main/jest.config.js#L2